src/app/services/templates.service.ts

```diff
@@ -136,6 +136,19 @@ export class TemplatesService {
             from(observables).pipe(
                 mergeAll(),
                 toArray(),
+                map(templateDescriptions => {
+                    var res: TemplateDescription[] = [];
+                    var idsAdded: string[] = [];
```

Comment:
Use a set. Sets have better Performance for this use-case.
Checking if Item is in Set: O(1)
Checking if Item is in List: O(n)
While you won't notice the difference with small n like here, it is always good to know which datastructure is suitable for which purpose.

Reply:
Also either use `let` or `const` in javascript and avoid `var` entirely! (var has weird scoping behaviour...)